platform/android/src/test/benchmark_runner.cpp

```diff
@@ -1,10 +1,312 @@
+    JavaWrapper<jclass> fileClass(env, env->FindClass("java/io/File"));
+    jmethodID fileCtor = env->GetMethodID(fileClass.get(), "<init>", "(Ljava/lang/String;Ljava/lang/String;)V");
+    jmethodID fileExists = env->GetMethodID(fileClass.get(), "exists", "()Z");
+
+    JavaWrapper<jstring> destination(env, env->NewStringUTF(destinationPath.c_str()));
+    JavaWrapper<jstring> file(env, env->NewStringUTF(fileName.c_str()));
+    JavaWrapper<jobject> fileToCopy(env, env->NewObject(fileClass.get(), fileCtor, destination.get(), file.get()));
+
+    bool stateOk = true;
+    if (env->CallBooleanMethod(fileToCopy.get(), fileExists)) {
+        mbgl::Log::Warning(mbgl::Event::General, "File '%s' already exists", filePath.c_str());
+    } else {
+        std::unique_ptr<AAsset, std::function<void(AAsset*)>> fileAsset(
+            AAssetManager_open(assetManager, fileName.c_str(), AASSET_MODE_BUFFER),
+            [](AAsset* asset) { AAsset_close(asset); });
+        if (fileAsset == nullptr) {
+            mbgl::Log::Warning(mbgl::Event::General, "Failed to open asset file %s", fileName.c_str());
+            return false;
+        }
+        const void* fileData = AAsset_getBuffer(fileAsset.get());
+        const off_t fileLen = AAsset_getLength(fileAsset.get());
+
+        std::unique_ptr<FILE, std::function<void(FILE*)>> newFile(std::fopen(filePath.c_str(), "w+"),
+                                                                  [](FILE* file) { std::fclose(file); });
+        stateOk = newFile != nullptr;
+
+        if (!stateOk) {
+            mbgl::Log::Warning(mbgl::Event::General, "Failed to create new file entry %s", fileName.c_str());
+        } else {
+            auto res = static_cast<off_t>(std::fwrite(fileData, sizeof(char), fileLen, newFile.get()));
+            if (fileLen != res) {
+                mbgl::Log::Warning(
+                    mbgl::Event::General, "Failed to generate file entry %s from assets", fileName.c_str());
+            }
+        }
+    }
+    return stateOk;
+}
+
+std::string jstringToStdString(JNIEnv* env, jstring jStr) {
+    if (!jStr) {
+        return std::string();
+    }
+
+    JavaWrapper<jclass> stringClass(env, env->GetObjectClass(jStr));
+    const jmethodID getBytes = env->GetMethodID(stringClass.get(), "getBytes", "(Ljava/lang/String;)[B");
+    JavaWrapper<jbyteArray> stringJbytes(
+        env,
+        static_cast<jbyteArray>(
+            env->CallObjectMethod(jStr, getBytes, JavaWrapper<jstring>(env, env->NewStringUTF("UTF-8")).get())));
+
+    size_t length = static_cast<size_t>(env->GetArrayLength(stringJbytes.get()));
+    jbyte* pBytes = env->GetByteArrayElements(stringJbytes.get(), NULL);
+
+    std::string ret = std::string(reinterpret_cast<char*>(pBytes), length);
+    env->ReleaseByteArrayElements(stringJbytes.get(), pBytes, JNI_ABORT);
+
+    return ret;
+}
+
+void unZipFile(JNIEnv* env, const std::string& zipFilePath, const std::string& destinationPath) {
```

Comment:
This unzip file is the same as the code on test runner, and probably will get duplicated again on the unit-test runner. I think what we need is a "android test runtime". Meaning a shared wrapper for render tests, unit tests and benchmark tests.

Reply:
I agree, I actually already have a patch for that but was planning to upload it as a separate PR.